packages/starknet-snap/src/utils/token.ts

```diff
@@ -0,0 +1,35 @@
+import { Config } from '../config';
+import { isSameChainId } from './snapUtils';
+
+/**
+ * Check if the token is preloaded token.
+ *
+ * @param params - The params object.
+ * @param params.tokenName - The token name.
+ * @param params.chainId - The chain id.
+ * @param params.tokenSymbol - The token symbol.
+ * @param params.tokenAddress - The token address.
+ * @returns True if the token is preloaded token, false otherwise.
+ */
+export function isPreloadedToken({
+  tokenName,
+  tokenSymbol,
+  tokenAddress,
+  chainId,
+}: {
+  tokenName: string;
+  tokenSymbol: string;
+  tokenAddress: string;
+  chainId: string;
+}) {
+  const bigIntTokenAddress = BigInt(tokenAddress);
+  return Boolean(
+    Config.preloadTokens.find(
```

Comment:
Tokens should be uniquely identified by their `address` and `chainId`, not by `name` or `symbol`.
In the ERC20 standard, multiple tokens can share the same `name` or `symbol`, since these are derived from the token contract.
Therefore, when determining token uniqueness, it's essential to rely on `address` and `chainId` alone.

Wdyt ?

Reply:
i think that make sense, as i copy the previous behaviour, didnt aware that, but that since our UI does not show the token address, so user can add 2 ETH with different address,

may be this is the reason we checking that way